src/utils/dayViewLayout.js

```diff
@@ -355,10 +358,56 @@ export function getStyledAvailabilities ({
     totalMin,
   };
 
-  availabilities.forEach((availability, availabilityIdx) => {
-    let style = getYStyles(availabilityIdx, helperArgs);
-    styledAvailabilities.push({ availability, style });
-  })
+  const overlapMap = {};
+  let columnIndex = 0;
+
+  for (let availabilityIndex = 0; availabilityIndex < availabilities.length; availabilityIndex++) {
+    const availability = availabilities[availabilityIndex];
+    let grouped = false;
+
+    // Check for overlap with existing groups
+    const overlapKeys = Object.keys(overlapMap);
+    for (let groupIndex = 0; groupIndex < overlapKeys.length; groupIndex++) {
+      const groupKey = overlapKeys[groupIndex];
+      const group = overlapMap[groupKey];
+      const lastAvailability = group[group.length - 1];
+      const lastAvailabilityEndTime = get(lastAvailability, availabilityEndAccessor);
+      const availabilityStartTime = get(availability, availabilityStartAccessor);
+      if (lastAvailabilityEndTime <= availabilityStartTime) {
+        // No overlap, add to current group
+        group.push(availability);
+        grouped = true;
+        break;
+      }
+    }
+
+    if (!grouped) {
+      // Create a new group
+      overlapMap[columnIndex] = [availability];
+      columnIndex++;
+    }
+  }
+
+  const overlapKeys = Object.keys(overlapMap);
+  for (let groupIndex = 0; groupIndex < overlapKeys.length; groupIndex++) {
+    const columnIndex = overlapKeys[groupIndex];
+    const group = overlapMap[columnIndex];
```

Comment:
Can we just do something like `Object.entries(overlapMap).forEach(...)` at the top level of the function instead? There's a lot of "declare variable and assign it the value of whatever's at the current index being iterated over" in this function that feels like clutter and makes the code harder to follow.
```tsx
// skip declaring `overlapKeys`, not needed

// `columnIndex` and `group` are both available as args for the `forEach` function
Object.entries(overlapMap).forEach(([columnIndex, group]) => {
  // ...
});